Failed to compile on Illumos-a0570dbda2 March 2021

Tried to compile htop from source 'htop 3.0.6-dev' on OpenIndiana 2020.10 build Illumos-a0570dbda2.
Configure detects OpenSolaris correctly and does not see any error here.

In the following the logs from configure and make.

- configure:

```
$ ./configure

checking build system type... i386-pc-solaris2.11
checking host system type... i386-pc-solaris2.11
checking target system type... i386-pc-solaris2.11
checking for a BSD-compatible install... /usr/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/gmkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /usr/bin/ggrep
checking for egrep... /usr/bin/ggrep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for gcc option to accept ISO C99... none needed
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for ANSI C header files... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for strings.h... (cached) yes
checking sys/param.h usability... yes
checking sys/param.h presence... yes
checking for sys/param.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking sys/utsname.h usability... yes
checking sys/utsname.h presence... yes
checking for sys/utsname.h... yes
checking for unistd.h... (cached) yes
checking whether sys/types.h defines makedev... no
checking sys/mkdev.h usability... yes
checking sys/mkdev.h presence... yes
checking for sys/mkdev.h... yes
checking execinfo.h usability... yes
checking execinfo.h presence... yes
checking for execinfo.h... yes
checking for pid_t... yes
checking for uid_t in sys/types.h... yes
checking for uint8_t... yes
checking for uint16_t... yes
checking for uint32_t... yes
checking for uint64_t... yes
checking for ceil in -lm... yes
checking for library containing kstat_open... -lkstat
checking for library containing Pgrab_error... -lproc
checking for library containing free... none required
checking for library containing clock_gettime... none required
checking for clock_gettime... yes
checking for faccessat... yes
checking for fstatat... yes
checking for host_get_clock_service... no
checking for memfd_create... no
checking for openat... yes
checking for readlinkat... yes
checking for library containing backtrace... none required
checking for addnwstr in -lncursesw6... no
checking for addnwstr in -lncursesw... yes
checking ncursesw/curses.h usability... no
checking ncursesw/curses.h presence... no
checking for ncursesw/curses.h... no
checking ncurses/ncurses.h usability... yes
checking ncurses/ncurses.h presence... yes
checking for ncurses/ncurses.h... yes
checking for library containing keypad... none required
checking for usable sched_setaffinity... no
checking for cap_init in -lcap... no
checking sys/capability.h usability... no
checking sys/capability.h presence... no
checking for sys/capability.h... no
configure: Linux delay accounting support can not be enabled, cause pkg-config is required for checking its availability
checking sensors/sensors.h usability... no
checking sensors/sensors.h presence... no
checking for sensors/sensors.h... no
checking whether C compiler accepts -Wnull-dereference... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating htop.1
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands

  htop 3.0.6-dev

  platform:                  solaris
  os-release file:           /etc/os-release
  (Linux) proc directory:    /proc
  (Linux) openvz:            no
  (Linux) vserver:           no
  (Linux) ancient vserver:   no
  (Linux) affinity:          no
  (Linux) delay accounting:  no
  (Linux) sensors:           no
  (Linux) capabilities:      no
  unicode:                   yes
  hwloc:                     no
  debug:                     no
  static:                    no
```

- make:
```
$ make

make  all-am
depbase=`echo solaris/Platform.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.  -DNDEBUG   -Wall -Wcast-align -Wcast-qual -Wextra -Wfloat-equal -Wformat=2 -Winit-self -Wmissing-format-attribute -Wmissing-noreturn -Wmissing-prototypes -Wpointer-arith -Wshadow -Wstrict-prototypes -Wundef -Wunused -Wwrite-strings -Wnull-dereference -pedantic -std=c99 -D_XOPEN_SOURCE_EXTENDED  -DSYSCONFDIR="\"/usr/local/etc\""  -I"./solaris" -g -O2 -MT solaris/Platform.o -MD -MP -MF $depbase.Tpo -c -o solaris/Platform.o solaris/Platform.c &&\
mv -f $depbase.Tpo $depbase.Po
In file included from /usr/include/sys/lwp.h:34:0,
                 from /usr/include/proc_service.h:43,
                 from /usr/include/libproc.h:47,
                 from solaris/Platform.h:12,
                 from solaris/Platform.c:10:
/usr/include/sys/ucontext.h:85:2: error: unknown type name 'stack_t'
  stack_t  uc_stack;
  ^~~~~~~
In file included from /usr/include/sys/thread.h:37:0,
                 from /usr/include/sys/procfs.h:70,
                 from /usr/include/procfs.h:47,
                 from /usr/include/libproc.h:49,
                 from solaris/Platform.h:12,
                 from solaris/Platform.c:10:
/usr/include/sys/klwp.h:116:2: error: unknown type name 'stack_t'
  stack_t lwp_sigaltstack; /* alternate signal stack */
  ^~~~~~~
/usr/include/sys/klwp.h:119:2: error: unknown type name 'k_sigset_t'
  k_sigset_t lwp_sigoldmask; /* for sigsuspend */
  ^~~~~~~~~~
In file included from /usr/include/sys/procfs.h:70:0,
                 from /usr/include/procfs.h:47,
                 from /usr/include/libproc.h:49,
                 from solaris/Platform.h:12,
                 from solaris/Platform.c:10:
/usr/include/sys/thread.h:171:2: error: unknown type name 'k_sigset_t'
  k_sigset_t t_sig;  /* signals pending to this process */
  ^~~~~~~~~~
/usr/include/sys/thread.h:172:2: error: unknown type name 'k_sigset_t'
  k_sigset_t t_extsig; /* signals sent from another contract */
  ^~~~~~~~~~
/usr/include/sys/thread.h:173:2: error: unknown type name 'k_sigset_t'
  k_sigset_t t_hold;  /* hold signal bit mask */
  ^~~~~~~~~~
/usr/include/sys/thread.h:174:2: error: unknown type name 'k_sigset_t'
  k_sigset_t t_sigwait; /* sigtimedwait/sigfd accepting these */
  ^~~~~~~~~~
In file included from /usr/include/procfs.h:47:0,
                 from /usr/include/libproc.h:49,
                 from solaris/Platform.h:12,
                 from solaris/Platform.c:10:
/usr/include/sys/procfs.h:135:19: error: field 'pr_action' has incomplete type
  struct sigaction pr_action; /* signal action for current signal */
                   ^~~~~~~~~
/usr/include/sys/procfs.h:136:2: error: unknown type name 'stack_t'
  stack_t pr_altstack; /* alternate signal stack info */
  ^~~~~~~
/usr/include/sys/procfs.h:530:2: error: unknown type name 'ino64_t'
  ino64_t  pr_ino;
  ^~~~~~~
/usr/include/sys/procfs.h:531:2: error: unknown type name 'off64_t'
  off64_t  pr_offset;
  ^~~~~~~
/usr/include/sys/procfs.h:532:2: error: unknown type name 'off64_t'
  off64_t  pr_size;
  ^~~~~~~
/usr/include/sys/procfs.h:550:2: error: unknown type name 'ino64_t'
  ino64_t  pr_ino;  /* inode number */
  ^~~~~~~
/usr/include/sys/procfs.h:551:2: error: unknown type name 'off64_t'
  off64_t  pr_size; /* file size */
  ^~~~~~~
/usr/include/sys/procfs.h:552:2: error: unknown type name 'off64_t'
  off64_t  pr_offset; /* current offset of file descriptor */
  ^~~~~~~
In file included from /usr/include/libctf.h:47:0,
                 from /usr/include/libproc.h:52,
                 from solaris/Platform.h:12,
                 from solaris/Platform.c:10:
/usr/include/sys/ctf_api.h:134:2: error: unknown type name 'off64_t'
  off64_t cts_offset; /* file offset of this section (if any) */
  ^~~~~~~
In file included from solaris/Platform.h:12:0,
                 from solaris/Platform.c:10:
/usr/include/libproc.h:332:42: error: unknown type name 'stack_t'
 extern int Lstack(struct ps_lwphandle *, stack_t *);
                                          ^~~~~~~
/usr/include/libproc.h:333:47: error: unknown type name 'stack_t'
 extern int Lmain_stack(struct ps_lwphandle *, stack_t *);
                                               ^~~~~~~
/usr/include/libproc.h:334:46: error: unknown type name 'stack_t'
 extern int Lalt_stack(struct ps_lwphandle *, stack_t *);
                                              ^~~~~~~
/usr/include/libproc.h:445:56: error: unknown type name 'stack_t'
 extern int Plwp_stack(struct ps_prochandle *, lwpid_t, stack_t *);
                                                        ^~~~~~~
/usr/include/libproc.h:446:61: error: unknown type name 'stack_t'
 extern int Plwp_main_stack(struct ps_prochandle *, lwpid_t, stack_t *);
                                                             ^~~~~~~
/usr/include/libproc.h:447:60: error: unknown type name 'stack_t'
 extern int Plwp_alt_stack(struct ps_prochandle *, lwpid_t, stack_t *);
                                                            ^~~~~~~
In file included from /usr/include/sys/proc.h:37:0,
                 from solaris/Platform.h:16,
                 from solaris/Platform.c:10:
/usr/include/sys/user.h:304:18: error: 'MAXSIG' undeclared here (not in a function)
  void (*u_signal[MAXSIG])(); /* Disposition of signals */
                  ^~~~~~
In file included from solaris/Platform.h:16:0,
                 from solaris/Platform.c:10:
/usr/include/sys/proc.h:200:2: error: unknown type name 'k_sigset_t'
  k_sigset_t p_sig;  /* signals pending to this process */
  ^~~~~~~~~~
/usr/include/sys/proc.h:201:2: error: unknown type name 'k_sigset_t'
  k_sigset_t p_extsig;  /* signals sent from another contract */
  ^~~~~~~~~~
/usr/include/sys/proc.h:202:2: error: unknown type name 'k_sigset_t'
  k_sigset_t p_ignore;  /* ignore when generated */
  ^~~~~~~~~~
/usr/include/sys/proc.h:203:2: error: unknown type name 'k_sigset_t'
  k_sigset_t p_siginfo;  /* gets signal info with signal */
  ^~~~~~~~~~
/usr/include/sys/proc.h:243:2: error: unknown type name 'k_sigset_t'
  k_sigset_t p_sigmask;  /* mask of traced signals (/proc) */
  ^~~~~~~~~~
/usr/include/sys/proc.h:335:2: error: unknown type name 'rlim64_t'
  rlim64_t p_stk_ctl; /* currently enforced stack size */
  ^~~~~~~~
/usr/include/sys/proc.h:336:2: error: unknown type name 'rlim64_t'
  rlim64_t p_fsz_ctl; /* currently enforced file size */
  ^~~~~~~~
/usr/include/sys/proc.h:337:2: error: unknown type name 'rlim64_t'
  rlim64_t p_vmem_ctl; /* currently enforced addr-space size */
  ^~~~~~~~
/usr/include/sys/proc.h:338:2: error: unknown type name 'rlim64_t'
  rlim64_t p_fno_ctl; /* currently enforced file-desc limit */
  ^~~~~~~~
In file included from ./Object.h:17:0,
                 from ./ListItem.h:12,
                 from ./Meter.h:16,
                 from ./Header.h:10,
                 from ./Action.h:15,
                 from solaris/Platform.h:19,
                 from solaris/Platform.c:10:
./XUtils.h: In function 'String_contains_i':
./XUtils.h:43:11: warning: implicit declaration of function 'strcasestr' [-Wimplicit-function-declaration]
    return strcasestr(s1, s2) != NULL;
           ^~~~~~~~~~
./XUtils.h:43:30: warning: comparison between pointer and integer
    return strcasestr(s1, s2) != NULL;
                              ^~
solaris/Platform.c: In function 'Platform_getMaxPid':
solaris/Platform.c:165:44: warning: passing argument 2 of 'kstat_lookup' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
       kstat_t* kshandle = kstat_lookup(kc, "unix", 0, "var");
                                            ^~~~~~
In file included from solaris/SolarisProcessList.h:18:0,
                 from solaris/Platform.c:27:
/usr/include/kstat.h:54:18: note: expected 'char *' but argument is of type 'const char *'
 extern kstat_t  *kstat_lookup(kstat_ctl_t *, char *, int, char *);
                  ^~~~~~~~~~~~
solaris/Platform.c:165:55: warning: passing argument 4 of 'kstat_lookup' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
       kstat_t* kshandle = kstat_lookup(kc, "unix", 0, "var");
                                                       ^~~~~
In file included from solaris/SolarisProcessList.h:18:0,
                 from solaris/Platform.c:27:
/usr/include/kstat.h:54:18: note: expected 'char *' but argument is of type 'const char *'
 extern kstat_t  *kstat_lookup(kstat_ctl_t *, char *, int, char *);
                  ^~~~~~~~~~~~
solaris/Platform.c: In function 'Platform_buildenv':
solaris/Platform.c:256:4: warning: implicit declaration of function 'strlcpy' [-Wimplicit-function-declaration]
    strlcpy( accump->env + accump->size, str, (accump->capacity - accump->size));
    ^~~~~~~
solaris/Platform.c: In function 'Platform_getProcessEnv':
solaris/Platform.c:269:14: warning: return discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
       return "Unable to read process environment.";
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*** Error code 1
make: Fatal error: Command failed for target `solaris/Platform.o'
Current working directory /tank/htop/htop
*** Error code 1
make: Fatal error: Command failed for target `all'
```

`stack_t`

```
typedef struct ip_stack ip_stack_t;
typedef int ip_stack_t;
typedef struct tcp_stack tcp_stack_t;
typedef int     mod_prop_setf_t(netstack_t *, cred_t *, mod_prop_info_t *,
typedef int     mod_prop_getf_t(netstack_t *, mod_prop_info_t *, const char *,
typedef struct str_stack str_stack_t;
typedef struct hook_stack hook_stack_t;
typedef struct neti_stack_s neti_stack_t;
typedef struct netstack netstack_t;
typedef volatile AO_double_t AO_stack_t;
```

`k_sigset_t`

notihng containing typedef 

`ino64_t`

```
typedef ino_t           ino64_t;        /* expanded inode type */
typedef u_longlong_t    ino64_t;        /* expanded inode type  */
typedef int (*gfs_lookup_cb)(vnode_t *, const char *, vnode_t **, ino64_t *,
typedef ino64_t (*gfs_inode_cb)(vnode_t *, int);
```

`off64_t`

```
typedef off_t           off64_t;
typedef __longlong_t    off64_t;
typedef off_t           off64_t;        /* offsets within files */
typedef longlong_t      off64_t;        /* offsets within files */
```

`rlim64_t`

```
typedef u_longlong_t    rlim64_t;
```

- make -k

```
make  all-am
depbase=`echo solaris/Platform.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.  -DNDEBUG   -Wall -Wcast-align -Wcast-qual -Wextra -Wfloat-equal -Wformat=2 -Winit-self -Wmissing-format-attribute -Wmissing-noreturn -Wmissing-prototypes -Wpointer-arith -Wshadow -Wstrict-prototypes -Wundef -Wunused -Wwrite-strings -Wnull-dereference -pedantic -std=c99 -D_XOPEN_SOURCE_EXTENDED  -DSYSCONFDIR="\"/usr/local/etc\""  -I"./solaris" -g -O2 -MT solaris/Platform.o -MD -MP -MF $depbase.Tpo -c -o solaris/Platform.o solaris/Platform.c &&\
mv -f $depbase.Tpo $depbase.Po
In file included from /usr/include/sys/lwp.h:34:0,
                 from /usr/include/proc_service.h:43,
                 from /usr/include/libproc.h:47,
                 from solaris/Platform.h:12,
                 from solaris/Platform.c:10:
/usr/include/sys/ucontext.h:85:2: error: unknown type name 'stack_t'
  stack_t  uc_stack;
  ^~~~~~~
In file included from /usr/include/sys/thread.h:37:0,
                 from /usr/include/sys/procfs.h:70,
                 from /usr/include/procfs.h:47,
                 from /usr/include/libproc.h:49,
                 from solaris/Platform.h:12,
                 from solaris/Platform.c:10:
/usr/include/sys/klwp.h:116:2: error: unknown type name 'stack_t'
  stack_t lwp_sigaltstack; /* alternate signal stack */
  ^~~~~~~
/usr/include/sys/klwp.h:119:2: error: unknown type name 'k_sigset_t'
  k_sigset_t lwp_sigoldmask; /* for sigsuspend */
  ^~~~~~~~~~
In file included from /usr/include/sys/procfs.h:70:0,
                 from /usr/include/procfs.h:47,
                 from /usr/include/libproc.h:49,
                 from solaris/Platform.h:12,
                 from solaris/Platform.c:10:
/usr/include/sys/thread.h:171:2: error: unknown type name 'k_sigset_t'
  k_sigset_t t_sig;  /* signals pending to this process */
  ^~~~~~~~~~
/usr/include/sys/thread.h:172:2: error: unknown type name 'k_sigset_t'
  k_sigset_t t_extsig; /* signals sent from another contract */
  ^~~~~~~~~~
/usr/include/sys/thread.h:173:2: error: unknown type name 'k_sigset_t'
  k_sigset_t t_hold;  /* hold signal bit mask */
  ^~~~~~~~~~
/usr/include/sys/thread.h:174:2: error: unknown type name 'k_sigset_t'
  k_sigset_t t_sigwait; /* sigtimedwait/sigfd accepting these */
  ^~~~~~~~~~
In file included from /usr/include/procfs.h:47:0,
                 from /usr/include/libproc.h:49,
                 from solaris/Platform.h:12,
                 from solaris/Platform.c:10:
/usr/include/sys/procfs.h:135:19: error: field 'pr_action' has incomplete type
  struct sigaction pr_action; /* signal action for current signal */
                   ^~~~~~~~~
/usr/include/sys/procfs.h:136:2: error: unknown type name 'stack_t'
  stack_t pr_altstack; /* alternate signal stack info */
  ^~~~~~~
/usr/include/sys/procfs.h:530:2: error: unknown type name 'ino64_t'
  ino64_t  pr_ino;
  ^~~~~~~
/usr/include/sys/procfs.h:531:2: error: unknown type name 'off64_t'
  off64_t  pr_offset;
  ^~~~~~~
/usr/include/sys/procfs.h:532:2: error: unknown type name 'off64_t'
  off64_t  pr_size;
  ^~~~~~~
/usr/include/sys/procfs.h:550:2: error: unknown type name 'ino64_t'
  ino64_t  pr_ino;  /* inode number */
  ^~~~~~~
/usr/include/sys/procfs.h:551:2: error: unknown type name 'off64_t'
  off64_t  pr_size; /* file size */
  ^~~~~~~
/usr/include/sys/procfs.h:552:2: error: unknown type name 'off64_t'
  off64_t  pr_offset; /* current offset of file descriptor */
  ^~~~~~~
In file included from /usr/include/libctf.h:47:0,
                 from /usr/include/libproc.h:52,
                 from solaris/Platform.h:12,
                 from solaris/Platform.c:10:
/usr/include/sys/ctf_api.h:134:2: error: unknown type name 'off64_t'
  off64_t cts_offset; /* file offset of this section (if any) */
  ^~~~~~~
In file included from solaris/Platform.h:12:0,
                 from solaris/Platform.c:10:
/usr/include/libproc.h:332:42: error: unknown type name 'stack_t'
 extern int Lstack(struct ps_lwphandle *, stack_t *);
                                          ^~~~~~~
/usr/include/libproc.h:333:47: error: unknown type name 'stack_t'
 extern int Lmain_stack(struct ps_lwphandle *, stack_t *);
                                               ^~~~~~~
/usr/include/libproc.h:334:46: error: unknown type name 'stack_t'
 extern int Lalt_stack(struct ps_lwphandle *, stack_t *);
                                              ^~~~~~~
/usr/include/libproc.h:445:56: error: unknown type name 'stack_t'
 extern int Plwp_stack(struct ps_prochandle *, lwpid_t, stack_t *);
                                                        ^~~~~~~
/usr/include/libproc.h:446:61: error: unknown type name 'stack_t'
 extern int Plwp_main_stack(struct ps_prochandle *, lwpid_t, stack_t *);
                                                             ^~~~~~~
/usr/include/libproc.h:447:60: error: unknown type name 'stack_t'
 extern int Plwp_alt_stack(struct ps_prochandle *, lwpid_t, stack_t *);
                                                            ^~~~~~~
In file included from /usr/include/sys/proc.h:37:0,
                 from solaris/Platform.h:16,
                 from solaris/Platform.c:10:
/usr/include/sys/user.h:304:18: error: 'MAXSIG' undeclared here (not in a function)
  void (*u_signal[MAXSIG])(); /* Disposition of signals */
                  ^~~~~~
In file included from solaris/Platform.h:16:0,
                 from solaris/Platform.c:10:
/usr/include/sys/proc.h:200:2: error: unknown type name 'k_sigset_t'
  k_sigset_t p_sig;  /* signals pending to this process */
  ^~~~~~~~~~
/usr/include/sys/proc.h:201:2: error: unknown type name 'k_sigset_t'
  k_sigset_t p_extsig;  /* signals sent from another contract */
  ^~~~~~~~~~
/usr/include/sys/proc.h:202:2: error: unknown type name 'k_sigset_t'
  k_sigset_t p_ignore;  /* ignore when generated */
  ^~~~~~~~~~
/usr/include/sys/proc.h:203:2: error: unknown type name 'k_sigset_t'
  k_sigset_t p_siginfo;  /* gets signal info with signal */
  ^~~~~~~~~~
/usr/include/sys/proc.h:243:2: error: unknown type name 'k_sigset_t'
  k_sigset_t p_sigmask;  /* mask of traced signals (/proc) */
  ^~~~~~~~~~
/usr/include/sys/proc.h:335:2: error: unknown type name 'rlim64_t'
  rlim64_t p_stk_ctl; /* currently enforced stack size */
  ^~~~~~~~
/usr/include/sys/proc.h:336:2: error: unknown type name 'rlim64_t'
  rlim64_t p_fsz_ctl; /* currently enforced file size */
  ^~~~~~~~
/usr/include/sys/proc.h:337:2: error: unknown type name 'rlim64_t'
  rlim64_t p_vmem_ctl; /* currently enforced addr-space size */
  ^~~~~~~~
/usr/include/sys/proc.h:338:2: error: unknown type name 'rlim64_t'
  rlim64_t p_fno_ctl; /* currently enforced file-desc limit */
  ^~~~~~~~
In file included from ./Object.h:17:0,
                 from ./ListItem.h:12,
                 from ./Meter.h:16,
                 from ./Header.h:10,
                 from ./Action.h:15,
                 from solaris/Platform.h:19,
                 from solaris/Platform.c:10:
./XUtils.h: In function 'String_contains_i':
./XUtils.h:43:11: warning: implicit declaration of function 'strcasestr' [-Wimplicit-function-declaration]
    return strcasestr(s1, s2) != NULL;
           ^~~~~~~~~~
./XUtils.h:43:30: warning: comparison between pointer and integer
    return strcasestr(s1, s2) != NULL;
                              ^~
solaris/Platform.c: In function 'Platform_getMaxPid':
solaris/Platform.c:165:44: warning: passing argument 2 of 'kstat_lookup' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
       kstat_t* kshandle = kstat_lookup(kc, "unix", 0, "var");
                                            ^~~~~~
In file included from solaris/SolarisProcessList.h:18:0,
                 from solaris/Platform.c:27:
/usr/include/kstat.h:54:18: note: expected 'char *' but argument is of type 'const char *'
 extern kstat_t  *kstat_lookup(kstat_ctl_t *, char *, int, char *);
                  ^~~~~~~~~~~~
solaris/Platform.c:165:55: warning: passing argument 4 of 'kstat_lookup' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
       kstat_t* kshandle = kstat_lookup(kc, "unix", 0, "var");
                                                       ^~~~~
In file included from solaris/SolarisProcessList.h:18:0,
                 from solaris/Platform.c:27:
/usr/include/kstat.h:54:18: note: expected 'char *' but argument is of type 'const char *'
 extern kstat_t  *kstat_lookup(kstat_ctl_t *, char *, int, char *);
                  ^~~~~~~~~~~~
solaris/Platform.c: In function 'Platform_buildenv':
solaris/Platform.c:256:4: warning: implicit declaration of function 'strlcpy' [-Wimplicit-function-declaration]
    strlcpy( accump->env + accump->size, str, (accump->capacity - accump->size));
    ^~~~~~~
solaris/Platform.c: In function 'Platform_getProcessEnv':
solaris/Platform.c:269:14: warning: return discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
       return "Unable to read process environment.";
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*** Error code 1
depbase=`echo solaris/SolarisProcess.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.  -DNDEBUG   -Wall -Wcast-align -Wcast-qual -Wextra -Wfloat-equal -Wformat=2 -Winit-self -Wmissing-format-attribute -Wmissing-noreturn -Wmissing-prototypes -Wpointer-arith -Wshadow -Wstrict-prototypes -Wundef -Wunused -Wwrite-strings -Wnull-dereference -pedantic -std=c99 -D_XOPEN_SOURCE_EXTENDED  -DSYSCONFDIR="\"/usr/local/etc\""  -I"./solaris" -g -O2 -MT solaris/SolarisProcess.o -MD -MP -MF $depbase.Tpo -c -o solaris/SolarisProcess.o solaris/SolarisProcess.c &&\
mv -f $depbase.Tpo $depbase.Po
In file included from ./Object.h:17:0,
                 from ./Process.h:15,
                 from solaris/SolarisProcess.c:9:
./XUtils.h: In function 'String_contains_i':
./XUtils.h:43:11: warning: implicit declaration of function 'strcasestr' [-Wimplicit-function-declaration]
    return strcasestr(s1, s2) != NULL;
           ^~~~~~~~~~
./XUtils.h:43:30: warning: comparison between pointer and integer
    return strcasestr(s1, s2) != NULL;
                              ^~
In file included from solaris/SolarisProcess.h:13:0,
                 from solaris/SolarisProcess.c:11:
/usr/include/sys/proc.h: At top level:
/usr/include/sys/proc.h:335:2: error: unknown type name 'rlim64_t'
  rlim64_t p_stk_ctl; /* currently enforced stack size */
  ^~~~~~~~
/usr/include/sys/proc.h:336:2: error: unknown type name 'rlim64_t'
  rlim64_t p_fsz_ctl; /* currently enforced file size */
  ^~~~~~~~
/usr/include/sys/proc.h:337:2: error: unknown type name 'rlim64_t'
  rlim64_t p_vmem_ctl; /* currently enforced addr-space size */
  ^~~~~~~~
/usr/include/sys/proc.h:338:2: error: unknown type name 'rlim64_t'
  rlim64_t p_fno_ctl; /* currently enforced file-desc limit */
  ^~~~~~~~
In file included from /usr/include/procfs.h:47:0,
                 from /usr/include/libproc.h:49,
                 from solaris/SolarisProcess.h:14,
                 from solaris/SolarisProcess.c:11:
/usr/include/sys/procfs.h:530:2: error: unknown type name 'ino64_t'
  ino64_t  pr_ino;
  ^~~~~~~
/usr/include/sys/procfs.h:531:2: error: unknown type name 'off64_t'
  off64_t  pr_offset;
  ^~~~~~~
/usr/include/sys/procfs.h:532:2: error: unknown type name 'off64_t'
  off64_t  pr_size;
  ^~~~~~~
/usr/include/sys/procfs.h:550:2: error: unknown type name 'ino64_t'
  ino64_t  pr_ino;  /* inode number */
  ^~~~~~~
/usr/include/sys/procfs.h:551:2: error: unknown type name 'off64_t'
  off64_t  pr_size; /* file size */
  ^~~~~~~
/usr/include/sys/procfs.h:552:2: error: unknown type name 'off64_t'
  off64_t  pr_offset; /* current offset of file descriptor */
  ^~~~~~~
In file included from /usr/include/libctf.h:47:0,
                 from /usr/include/libproc.h:52,
                 from solaris/SolarisProcess.h:14,
                 from solaris/SolarisProcess.c:11:
/usr/include/sys/ctf_api.h:134:2: error: unknown type name 'off64_t'
  off64_t cts_offset; /* file offset of this section (if any) */
  ^~~~~~~
solaris/SolarisProcess.c: In function 'SolarisProcess_writeField':
solaris/SolarisProcess.c:77:41: warning: format '%d' expects argument of type 'int', but argument 5 has type 'zoneid_t {aka const long int}' [-Wformat=]
    case ZONEID: xSnprintf(buffer, n, "%*d ", Process_pidDigits, sp->zoneid); break;
                                         ^
solaris/SolarisProcess.c:78:41: warning: format '%d' expects argument of type 'int', but argument 5 has type 'projid_t {aka const long int}' [-Wformat=]
    case PROJID: xSnprintf(buffer, n, "%*d ", Process_pidDigits, sp->projid); break;
                                         ^
solaris/SolarisProcess.c:79:41: warning: format '%d' expects argument of type 'int', but argument 5 has type 'taskid_t {aka const long int}' [-Wformat=]
    case TASKID: xSnprintf(buffer, n, "%*d ", Process_pidDigits, sp->taskid); break;
                                         ^
solaris/SolarisProcess.c:80:41: warning: format '%d' expects argument of type 'int', but argument 5 has type 'poolid_t {aka const long int}' [-Wformat=]
    case POOLID: xSnprintf(buffer, n, "%*d ", Process_pidDigits, sp->poolid); break;
                                         ^
solaris/SolarisProcess.c:81:41: warning: format '%d' expects argument of type 'int', but argument 5 has type 'ctid_t {aka const long int}' [-Wformat=]
    case CONTID: xSnprintf(buffer, n, "%*d ", Process_pidDigits, sp->contid); break;
                                         ^
solaris/SolarisProcess.c:83:38: warning: format '%d' expects argument of type 'int', but argument 5 has type 'pid_t {aka const long int}' [-Wformat=]
    case PID: xSnprintf(buffer, n, "%*d ", Process_pidDigits, sp->realpid); break;
                                      ^
solaris/SolarisProcess.c:84:39: warning: format '%d' expects argument of type 'int', but argument 5 has type 'pid_t {aka const long int}' [-Wformat=]
    case PPID: xSnprintf(buffer, n, "%*d ", Process_pidDigits, sp->realppid); break;
                                       ^
solaris/SolarisProcess.c:85:40: warning: format '%d' expects argument of type 'int', but argument 5 has type 'pid_t {aka const long int}' [-Wformat=]
    case LWPID: xSnprintf(buffer, n, "%*d ", Process_pidDigits, sp->lwpid); break;
                                        ^
*** Error code 1
make: Warning: Target `all-am' not remade because of errors
Current working directory /tank/htop/htop
*** Error code 1
make: Warning: Target `all' not remade because of errors
```

--- a/solaris/Platform.c
+++ b/solaris/Platform.c
@@ -7,7 +7,19 @@ Released under the GNU GPLv2, see the COPYING file
 in the source distribution for its full text.
 */
 
-#include "Platform.h"
+#include "solaris/Platform.h"
+
+#include <kstat.h>
+#include <math.h>
+#include <string.h>
+#include <time.h>
+#include <utmpx.h>
+#include <sys/loadavg.h>
+#include <sys/resource.h>
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/var.h>
+
 #include "Macros.h"
 #include "Meter.h"
 #include "CPUMeter.h"
@@ -25,17 +37,6 @@ in the source distribution for its full text.
 #include "zfs/ZfsCompressedArcMeter.h"
 #include "SolarisProcess.h"
 #include "SolarisProcessList.h"
-
-#include <sys/types.h>
-#include <sys/time.h>
-#include <sys/resource.h>
-#include <utmpx.h>
-#include <sys/loadavg.h>
-#include <string.h>
-#include <kstat.h>
-#include <time.h>
-#include <math.h>
-#include <sys/var.h>
 
 
 double plat_loadavg[3] = {0};

--- a/solaris/Platform.h
+++ b/solaris/Platform.h
@@ -9,9 +9,12 @@ Released under the GNU GPLv2, see the COPYING file
 in the source distribution for its full text.
 */
 
+#include "config.h" // IWYU pragma: keep
+
 #include <libproc.h>
 #include <signal.h>
 #include <stdbool.h>
+
 #include <sys/mkdev.h>
 #include <sys/proc.h>
 #include <sys/types.h>
@@ -22,6 +25,7 @@ in the source distribution for its full text.
 #include "NetworkIOMeter.h"
 #include "ProcessLocksScreen.h"
 #include "SignalsPanel.h"
+
 #include "generic/hostname.h"
 #include "generic/uname.h"
 

--- a/solaris/SolarisProcess.c
+++ b/solaris/SolarisProcess.c
@@ -6,16 +6,18 @@ Released under the GNU GPLv2, see the COPYING file
 in the source distribution for its full text.
 */
 
-#include "Process.h"
-#include "ProcessList.h"
-#include "SolarisProcess.h"
-#include "Platform.h"
-#include "CRT.h"
+#include "solaris/SolarisProcess.h"
 
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
 #include <sys/syscall.h>
+
+#include "Process.h"
+#include "ProcessList.h"
+#include "CRT.h"
+
+#include "solaris/Platform.h"
 
 
 const ProcessFieldData Process_fields[LAST_PROCESSFIELD] = {

--- a/solaris/SolarisProcess.h
+++ b/solaris/SolarisProcess.h
@@ -8,10 +8,14 @@ Released under the GNU GPLv2, see the COPYING file
 in the source distribution for its full text.
 */
 
-#include "Settings.h"
+#include "config.h" // IWYU pragma: keep
+
 #include <zone.h>
 #include <sys/proc.h>
 #include <libproc.h>
+
+#include "Settings.h"
+
 
 typedef struct SolarisProcess_ {
    Process    super;

